memcache/memcache.go

```diff
@@ -277,6 +282,10 @@ func (c *Client) getConn(addr net.Addr) (*conn, error) {
 	return cn, nil
 }
 
+func (c *Client) SetMaxIdleConnsPerAddr(n int) {
+	c.maxIdleConnsPerAddr = n
```

Comment:
LGTM, except that we may need a lock here.  If we do, then the change is big.  Maybe in the constructor?